### ISSUE TITLE
include shell in ShellCheck Docker image for glob support

### ISSUE THREAD

**Issue by brovoca** [koalaman/shellcheck#2862]
Hello ShellCheck team,

I am currently using the ShellCheck Docker image (version 0.9.0) in a Tekton pipeline and have encountered an issue that limits its usability. The image available at [Docker Hub](https://hub.docker.com/layers/koalaman/shellcheck/v0.9.0/images/sha256-a527e2077f11f28c1c1ad1dc784b5bc966baeb3e34ef304a0ffa72699b01ad9c?context=explore) is based on `scratch` and contains only the ShellCheck binary. This setup creates a problem when trying to use glob patterns for file selection.

Glob patterns are generally expanded by a shell, but in the context of Tekton pipelines, this does not happen. Arguments like `**/*sh` are passed as strings to the container and thus are not expanded. A potential solution would be to include a shell in the Docker image, which would enable executing ShellCheck through a script capable of expanding globs.

Here's an example of the issue:

```sh
docker run --rm koalaman/shellcheck:v0.9.0 "*"
# Outputs: *: *: openBinaryFile: does not exist (No such file or directory)
```

Attempting to use an entrypoint results in an error due to the absence of a shell:

```sh
docker run --rm --entrypoint sh koalaman/shellcheck:v0.9.0 
# Outputs: docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "sh": executable file not found in $PATH: unknown.
```

To enhance the functionality of the ShellCheck Docker image, especially for use cases like Tekton pipelines, I suggest including a minimal shell environment and common tools such as `read`, `echo`, `find`, `tr`, `printf` and `sort`. This addition would allow us to define a set of regex expressions to identify files for linting, as illustrated in the following script:

```sh
#!/bin/bash

# Imagine that this variable is set in the environment by Tekton.
# Space separated list of regexes that we'll loop over.
EXPRESSIONS=".*\.sh \./\.make/.* \./\.git/hooks/.*"

# Array of files found
files=()

# Loop over each expression
while read -r regex; do
    echo "regex: $regex"

    # Add all files matching expression to the array
    while read -r file; do
        files+=("$file")
    done <<<"$(find ./ -type f -regex "$regex")"
done <<<"$(tr ' ' '\n' <<<"$EXPRESSIONS")"

# Deduplicate list
deduplicated=()
while IFS= read -r -d '' x; do deduplicated+=("$x"); done < <(printf "%s\0" "${files[@]}" | sort -uz)

# Run check
shellcheck "${deduplicated[@]}"
```

This enhancement would greatly improve the flexibility and usefulness of the ShellCheck Docker image in CI/CD pipelines.

Thank you for considering this request.

**Comment by koalaman** [comment]
Check out `koalaman/shellcheck-alpine`, a base Alpine Linux distro with `shellcheck` included for situations where shellcheck by itself is too barebones. It comes with a basic shell and utilities, and Alpine's `apk` to install fancier shells like Bash.

**Comment by brovoca** [comment]
Ah, thanks!
